lib/fixed-points.nix

```diff
@@ -63,7 +63,19 @@ rec {
   #             = self: { foo = "foo"; bar = "bar"; foobar = self.foo + self.bar; } // { foo = "foo" + " + "; }
   #             = self: { foo = "foo + "; bar = "bar"; foobar = self.foo + self.bar; }
   #
-  extends = f: rattrs: self: let super = rattrs self; in super // f self super;
+  mergeUpdate = lhs: rhs:
+    let
+      merge = name: value:
+        if builtins.hasAttr name lhs
+            && (rhs."_merge_${name}" or lhs."_merge_${name}" or false)
```

Comment:
Could we have a function instead similar to `recurseIntoAttrs`?

Reply:
but then we have to define all sets in one place (in that function), right? In my implementation it is possible to hijack anything from overlay, without the need to extend function.

But maybe you can show an example of what you expect that "merge-approve" function should look like

Reply:
I mean something like value 

```nix
mergeInOverlays = attrs: attrs // { mergeInOverlays = true; };
mergeUpdate = lhs: rhs:
  # ...
  # this is incorrect because both the value from `lhs` and `rhs` must be checked, 
  # but the idea should be clear.
  if (builtins.isAttrs value && value ? mergeInOverlays)
  # ...

python2Packages = mergeInOverlays python2.pkgs;
```

Reply:
Unfortunately it is not possible, because it requires evaluate.of all attributes (we want to know which attributes are sets, and which of those have `mergeInOverlays` sub-attribute).

And some of our attributes rely on tying the knot, so we don't want to evaluate them too soon.

In particular, `wrapBintoolsWith` evaluation relies on `stdenv`, which itself relies on `binutils`, which relies on `wrapBintoolsWith`... I couldn't make it play.